### PR TITLE
Issue-612 BOOKKEEPER-818: Make bookie recovery work with recovering multiple bookies

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

This is a change ported from twitter branch.

- Enable recovering multiple bookies in bookie shell
- Add a unit test for BookieShell.

### CODE REVIEW

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Comment by asfgit** [comment]
FAILURE
 
--none--

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java [hunk 1]
```diff
@@ -442,6 +449,69 @@ public SyncObject() {
         }
     }
 
+    public SortedMap<Long, LedgerMetadata> getLedgersContainBookies(Set<BookieSocketAddress> bookies)
+            throws InterruptedException, BKException {
+        final SyncObject sync = new SyncObject();
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java]: I know this is kind of cherry pick.
Can you consider dropping this and use CompletableFuture. This legacy way of coding can be affected by spurious wake ups

Comment (sijie) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java]: I would defer this to a separate task. BookKeeperAdmin needs a cleanup. I don't want to mix two approaches in the same file. how does that sound?

Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java]: sure, it is best to make one step at a time

**Pull request closed without merging**